lib/idl.ml

```diff
@@ -22,15 +22,33 @@ module Error = struct
     matcher : exn -> 'a option;
   }
 
-  module Make(T : sig type t val t : t Rpc.Types.def end) = struct
-    exception Exn of T.t
-    let error = {
-        def = T.t;
-        raiser = (function e -> Exn e);
-        matcher = (function | Exn e -> Some e | _ -> None)
-      }
+  module type ERROR = sig
+    type t
+    val t : t Rpc.Types.def
   end
 
+  module type INTERNAL_ERROR = sig 
+    include ERROR
+    val internal_error_of: exn -> t option
+  end
```

Comment:
Let's not have the ERROR one, let's force everyone to declare how to handle unexpected exceptions.